Disable pprint line wrapping in the doctests

asmeurer opened the pull request:
This makes the doctests pass locally, as were currently hard-coded to whatever
column length was set on CI.

Note that single line outputs can still be manually line wrapped, as the
doctester ignores newlines. Multiline outputs will need to either be included
as the full length or manually wrapped by passing num_columns to pprint() (or
something other than pprint used).

Fixes #26318.<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed


#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

asmeurer committed:
Disable pprint line wrapping in the doctests

This makes the doctests pass locally, as were currently hard-coded to whatever
column length was set on CI.

Note that single line outputs can still be manually line wrapped, as the
doctester ignores newlines. Multiline outputs will need to either be included
as the full length or manually wrapped by passing num_columns to pprint() (or
something other than pprint used).

Fixes #26318.

asmeurer committed:
Fix some doctest formatting

asmeurer committed:
Fix line-wrapped pprint outputs in the doctests

asmeurer committed:
Skip failing doctests in beam.py

asmeurer commented:
I couldn't find anywhere that was setting `$COLUMNS` on CI. It must have just been using some default value. At any rate, I confirmed locally with my wide terminal window that the doctests now pass.

asmeurer commented:
Obviously there's a lot of related cleanups that could be made here. Many of these instances probably could just demonstrate the expression using inline LaTeX. Also I don't know if we should be using Unicode pprint anywhere but we apparently are (the formatting of those pages will look bad on computers without proper font support, see #15700). Also the continuation stuff itself is still up for discussion #26221. 

But we can fix these issues separately. My goal here was just to fix the doctest issues that were blocking the release.

oscarbenjamin commented:
> I couldn't find anywhere that was setting `$COLUMNS` on CI. It must have just been using some default value

Is it this:
https://github.com/sympy/sympy/blob/ae3027b4ca682023dee36e7e8883311c62cc08e3/sympy/printing/pretty/stringpict.py#L320-L324

That was recently modified in 992eaf56523 from gh-25753.

asmeurer commented:
Looks like doctests are failing on CI, though, so I'll need to figure that out.

asmeurer commented:
Perhaps I was being fooled by my terminal already being extra wide and my fix doesn't actually work.

oscarbenjamin commented:
> I just tried locally with a wide and then a narrow terminal and it seemed to work (with this PR).

I only tested the docstrings. For the rst docs it fails with pytest-doctestplus and also with bin/doctest.

asmeurer committed:
Properly disable line wrapping in the doctests

This can be checked with something like

COLUMNS=50 ./bin/doctest

This also removes the need to set wrap_line=False when calling init_printing()
from a doctest.

asmeurer commented:
I'm a little confused why this test passes on master, and why the changes here make it fail

```py
    huge_expr = 0
    for i in range(20):
        huge_expr += i*sin(i + x)
    assert xpretty(huge_expr            ).find('\n') != -1
```

If I run `pretty(huge_expr)` interactively, the default is to line wrap, so it does have newlines.

asmeurer commented:
Not sure what to do about pytest-doctestplus. In principle, it should just need to call `setup_pprint` similar to how the normal doctester does.

oscarbenjamin commented:
> Not sure what to do about pytest-doctestplus

We don't need to worry about pytest-doctestplus right now.

asmeurer committed:
Don't disable line wrapping in the tests

asmeurer commented:
I figured out the tests. 

The code here turning off and on global settings for the tests is a little messy. Any suggestions for improving it are welcome. We should probably unify the line wrapping and the unicode logic somehow.

asmeurer commented on conftest.py:
```diff
@@ -16,7 +16,7 @@
 
 # Set up printing for doctests
-setup_pprint()
+setup_pprint(disable_line_wrap=False)
```
Presumably pytest-doctestplus would actually do the right thing without this flag, but this also breaks the line pprint tests that test the line wrapping defaults.
